core/src/test/java/com/yahoo/imapnio/async/data/StoreResultTest.java

```diff
@@ -0,0 +1,93 @@
+package com.yahoo.imapnio.async.data;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.sun.mail.iap.ProtocolException;
+import com.sun.mail.imap.protocol.FetchResponse;
+import com.sun.mail.imap.protocol.IMAPResponse;
+import com.sun.mail.imap.protocol.MODSEQ;
+
+/**
+ * Unit test for {@code StoreResult}.
+ */
+public class StoreResultTest {
+    /**
+     * Tests StoreResult constructor and getters.
+     */
+    @Test
+    public void testStoreResult() throws IOException, ProtocolException {
+        final IMAPResponse[] content = new IMAPResponse[2];
+        content[0] = new IMAPResponse("* 1 FETCH (UID 4 MODSEQ (12121231000))");
+        content[1] = new IMAPResponse("* VANISHED (EARLIER) 41,43:116,118,120:211,214:540");
+        final List<IMAPResponse> imapResponses = Arrays.asList(content);
+        final MessageNumberSet[] modifiedMsgNums = { new MessageNumberSet(1L, 1L) };
+        final StoreResult storeResult = new StoreResult(1L, imapResponses, modifiedMsgNums);
+        final long highestModSeq = storeResult.getHighestModSeq();
+        final List<IMAPResponse> responses = storeResult.getIMAPResponses();
+
+        Assert.assertEquals(responses.size(), 2, "Result mismatched.");
+        Assert.assertEquals(highestModSeq, 1L, "Result mismatched.");
+        Assert.assertTrue(responses.get(0).keyEquals("FETCH"), "Result mismatched.");
+        final long fetchedModSeq = new FetchResponse(responses.get(0)).getItem(MODSEQ.class).modseq;
```

Comment:
LInes 37,38 is not in the scope of this UT, they are testing FetchResponse. You can remove them.